examples/healthData/src/translator.ts

```diff
@@ -0,0 +1,75 @@
+import {Result, TypeChatLanguageModel, createJsonTranslator, TypeChatJsonTranslator} from "typechat";
+
+type ChatMessage = {
+    source: "system" | "user" | "assistant";
+    body: string;
+};
+
+export interface TranslatorWithHistory<T extends object> {
+    _chatHistory: ChatMessage[];
+    _maxPromptLength: number;
+    _additionalAgentInstructions: string;
+    _translator: TypeChatJsonTranslator<T>;
+    translate(request: string): Promise<Result<T>>;
+}
+
+export function createHealthDataTranslator<T extends object>(model: TypeChatLanguageModel, schema: string, typename: string, additionalAgentInstructions: string): TranslatorWithHistory<T> {
+    const _chatHistory: ChatMessage[] = [];
+    const _maxPromptLength = 2048;
+    const _additionalAgentInstructions = additionalAgentInstructions;
+    
+    const _translator = createJsonTranslator<T>(model, schema, typename);
+    _translator.createRequestPrompt = createRequestPrompt;
+    
+    const customtranslator: TranslatorWithHistory<T> = {
+        _chatHistory,
+        _maxPromptLength,
+        _additionalAgentInstructions,
+        _translator,
+        translate,
+    };
+
+    return customtranslator;
+
+    async function translate(request: string): Promise<Result<T>> {
+        const response = await _translator.translate(request);
+        if (response.success) {
+            _chatHistory.push({ source: "assistant", body: JSON.stringify(response.data) });
```

Comment:
Can you explain to me why the body needs to be `stringified`?

For example, let's say your response was something like

```json
{ "foo": "bar" }
```

Your initial chat history would be represented in code as:

```json
[
    "{ \"foo\": \"bar\" }"
]
```

and eventually *that* gets stringified when you construct `historyStr`. So what's the rationale?

Reply:
type ChatMessage = {
    source: "system" | "user" | "assistant";
    body: object;
};

this fixes it.